### ISSUE TITLE
Enable turning off autoprefixer

### ISSUE THREAD

**Issue by m90** [sintaxi/harp#417]
While the autoprefixer definitely is great, it is a little too upfront for some of us in its assumptions regarding what to prefix / not to prefix.

The case I ran into right now was that it will turn this:

```less
.box(){
-webkit-box-sizing: border-box;
-moz-box-sizing: border-box;
box-sizing: border-box;
}
.foo-bar{
.box();
}
```
into
```css
.foo-bar{box-sizing:border-box}
```
While that's all fine in an evergreen world I do have clients using Android 2 browsers or some weird nightly build version of Firefox 17 that won't go away because the sysadmin says it won't. Meaning I do need the prefixed properties as well.

While that is my problem I think it would be nice to be enable people to turn off "modern" features like that.

**Comment by kennethormandy** [comment]
Hey @m90,

Thanks for opening an issue. This isn’t something we have any plans to make configurable, Harp aims to do the best job it can for the majority use cases by not exposing those options unless it’s absolutely necessary.

If you’d like to do something custom, I have a couple of recommendations:

1. Install Autoprefixer as well, and run it on your CSS files after running `harp compile` with a custom configuration that enables Android 2 and Firefox 17 (this should restore the prefixes)
2. Or, write your older styles in an old `.css` file. Harp only runs Autoprefixer on `.sass`, `.scss`, `.less` and `.styl` files, so your prefixes will remain intact there. Then you can use a traditional CSS import or conditional link tag in your markup to get this additional file.

Hope those are helpful for you, and feel free to post if you have any other questions.

**Comment by sintaxi** [comment]
Did some rough calculations...

- Gingerbread is currently at 6.9% of Android devices (but dropping fast).
- Android is accounts for roughly 50% of mobile.
- Mobile accounts for roughly 50% or usage vs Desktop.

Based on those rough calculations Gingerbread is still about 2% of the market, which is not insignificant. @kennethormandy can autoprefixer easily be configured to support more devices? I think I prefer that solution best.

Resources:
http://developer.android.com/about/dashboards/index.html
http://www.cnet.com/news/android-loses-some-us-market-share-but-remains-top-dog/
http://searchenginewatch.com/sew/opinion/2353616/mobile-now-exceeds-pc-the-biggest-shift-since-the-internet-began

**Comment by kennethormandy** [comment]
It’s completely configurable, but I think the best option in Harp is to use the defaults Autoprefixer comes with. The Autoprefixer maintainers have a process for deciding when to deprecate prefixes based on the CanIUse data, on a property-by-property basis.

To use the example provided, `border-radius` is actually supported [by Android 2.2 and higher](http://caniuse.com/#feat=border-radius) so that prefix has been dropped. You can override this, but then you lose out the expertise provided by the Autoprefixer team’s decisions and data for all of Android because you’ve held yourself to Android 2.1 support.

I think the best general configuration for Autopreifxer is the default, my concern is that if you’re doing a project that needs specific configuration, Harp’s settings would probably still be inappropriate. They just have to be dealt with on a project by project basis otherwise.

**Comment by m90** [comment]
Thanks for the whole lot of feedback. I totally can understand that you want to keep harp simple.

I already used the CSS import way you are describing before I wrote the issue so this was more of a meta question.

Just to add my 5 cents on market shares: While it's fine to use autoprefixer's approach in a pure "website world", your clients will break such metrics pretty quick. In my particular case I have a client where a ~500 employees corporation **has to** use some locked pre-release/nightly build version of Firefox 17. Until the end of days. Sysadmin will shout at you on the telephone when you try to convince him that might not be the best idea ever. So: you never know by just looking at the metrics.

In any case: I do love Harp a lot and will continue using it, yet as a developer I might wish for some secret backdoors to tweak the configuration from time to time. Thanks.
